lib/key-commands/index.ts

```diff
@@ -0,0 +1,278 @@
+import * as fs from "fs";
+import { platform as currentPlatform } from "os";
+import * as path from "path";
+import { color } from "../color";
+import { PrepareCommand } from "../commands/prepare";
+import { IChildProcess } from "../common/declarations";
+import { ICommand } from "../common/definitions/commands";
+import {
+	IKeyCommand,
+	IKeyCommandPlatform,
+	IValidKeyCommands,
+	SupportedProcessType,
+} from "../common/definitions/key-commands";
+import { injector } from "../common/yok";
+import { IProjectData } from "../definitions/project";
+import { IStartService } from "../definitions/start-service";
+
+export class A implements IKeyCommand {
+	key: IValidKeyCommands = "a";
+	platform: IKeyCommandPlatform = "Android";
+	description: string = "Run android app";
+
+	constructor(private $startService: IStartService) {}
+
+	async execute(): Promise<void> {
+		this.$startService.runAndroid();
+	}
+
+	canExecute(processType: SupportedProcessType) {
+		return processType === "start";
+	}
+}
+
+export class ShiftA implements IKeyCommand {
+	key: IValidKeyCommands = "A";
+	platform: IKeyCommandPlatform = "Android";
+	description: string = "Open android project in Android Studio";
+	willBlockKeyCommandExecution: boolean = true;
+
+	constructor(
+		private $logger: ILogger,
+		private $liveSyncCommandHelper: ILiveSyncCommandHelper,
+		private $childProcess: IChildProcess,
+		private $projectData: IProjectData
+	) {}
+
+	async execute(): Promise<void> {
+		this.$liveSyncCommandHelper.validatePlatform(this.platform);
+		this.$projectData.initializeProjectData();
+		const androidDir = `${this.$projectData.platformsDir}/android`;
+
+		if (!fs.existsSync(androidDir)) {
+			const prepareCommand = injector.resolveCommand(
+				"prepare"
+			) as PrepareCommand;
+			await prepareCommand.execute([this.platform]);
+			process.stdin.resume();
+		}
+
+		const os = currentPlatform();
+
+		if (os === "darwin") {
+			if (!fs.existsSync("/Applications/Android Studio.app")) {
+				this.$logger.error("Android Studio is not installed");
+				return;
+			}
+			this.$childProcess.exec(
+				`open -a "/Applications/Android Studio.app" ${androidDir}`
+			);
+		} else if (os === "win32") {
+			const studioPath = path.join(
+				"C:",
+				"Program Files",
+				"Android",
+				"Android Studio",
+				"bin",
+				"studio64.exe"
+			);
+			if (!fs.existsSync(studioPath)) {
+				this.$logger.error("Android Studio is not installed");
+				return;
+			}
+
+			this.$childProcess.exec(`${studioPath} ${androidDir}`);
```

Comment:
It needs to escape any spaces in the path by adding quotes, 
```suggestion
			this.$childProcess.exec(`'${studioPath}" "${androidDir}"`);
```